doc/source/reference/c-api.config.rst

```diff
@@ -80,6 +80,8 @@ Platform information
 
     Defined in ``numpy/npy_cpu.h``
 
+.. c:var:: NPY_HAVE_ENDIAN_H
+
```

Comment:
All the `NPY_HAVE_` macros are generated at build by probing the platform / compiler runtimes for features. If we wish to document them, we should somehow gather all the checks automatically from calls to [`numpy.distutils.command.config.config.check*`](https://github.com/numpy/numpy/blob/v1.16.3/numpy/distutils/command/config.py#L166)

Or just (for now) not try to document them :)

Reply:
hmm, not sure about this one either way - I see that they can be useful and I think it would work to use them from outside numpy, however they weren't really meant as public API

Reply:
@rgommers are you saying this should stay as-is?

Reply:
I'm saying I don't know. I *think* they weren't designed for outside usage. They *could* be useful for outside usage. I'm not aware of usage outside of numpy. A quick search in SciPy, scikit-learn, PyWavelets comes up empty. Usage of `NPY_HAVE_ENDIAN_H` can be found but are mostly partial or full copies of numpy build code:  (e.g. https://searchcode.com/file/12386275/src/package/extras/numpy_include/npy_endian.h) rather than standalone usages.

> Or just (for now) not try to document them :)

Agree that I would leave them out of this PR, and do either a standalone one that's complete or nothing. Not sure about it, what's your opinion?

Reply:
I would prefer to keep them out of this PR.